File: 141-LinkedListCycle/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        hashset = {}
        while head:
            if head in hashset:
                return True
            else:
                hashset[head] = True
            head = head.next
        return False

File: 141-LinkedListCycle/Solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        unordered_map<ListNode*,int> visited;
        while(head){
            if(visited.count(head)>0)
                return true;
            else
                visited[head]=1;
            head=head->next;
        }
        return false;
        }
};
